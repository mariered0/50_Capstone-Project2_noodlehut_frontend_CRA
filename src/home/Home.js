import React, { useContext } from "react";
import { Link } from "react-router-dom";
import userContext from "../user/UserContext";
import { Box, Container, Typography, Button, Grid, Stack } from "@mui/material";
import RestaurantIcon from "@mui/icons-material/Restaurant";
import HomeIcon from "@mui/icons-material/Home";
import ComputerIcon from "@mui/icons-material/Computer";
import heroImg from "../static/image/hero.jpg";

function Home() {
  const { currentUser } = useContext(userContext);

  return (
    
      <Container sx={{ pt: 6, pb: 6 }}>
        {currentUser ? (
          <Typography
            component="h3"
            variant="h3"
            align="center"
            color="textPrimary"
            gutterBottom
          >
            Welcome back, {currentUser?.firstName}!
          </Typography>
        ) : (
          ""
        )}

        <Box
          sx={{
            display: "flex",
            justifyContent: "center",
          }}
        >
          <Box
            component="img"
            sx={{
              height: 500,
              wedth: 650,
              maxHeight: { xs: 230, md: 500 },
              maxWidth: { xs: 350, md: 700 },
            }}
            alt="food image"
            src={heroImg}
          />
        </Box>

        <Typography
          component="h1"
          variant="h2"
          align="center"
          color="textPrimary"
          gutterBottom
          sx={{ pt: 3 }}
          className="title"
        >
          NoodleHut
        </Typography>

        <Typography
          variant="h5"
          align="center"
          color="text.secondary"
          paragraph
        >
          Thai home-style coocking. We serve authentically traditional Thai
          dishes.
          <br />
          No dine-in. This is a take-out restaurant.
        </Typography>

        {currentUser ? (
          ""
        ) : (
          <Link to="/menu">
            <Box textAlign="center">
              <Button variant="contained" startIcon={<RestaurantIcon />}>
                Care to order?
              </Button>
            </Box>
          </Link>
        )}

      <Box
        sx={{
          width: "100%",
          // display: "flex",
          // justifyContent: "space-around",
          mt: 4,
        }}
        component="footer"
        
      >
        <Grid
          container
          spacing={{ xs: 3, md: 5 }}
          columns={{ xs: 4, sm: 8, md: 12 }}
        >
          <Grid item sm={12} md={4} key={1} display="flex" justifyContent={"center"}>
              <Box>
              <Typography
                variant="h5"
                component="div"
                align="center"
                color="primary"
              >
                Visit
              </Typography>

              <Stack direction="row" alignItems="center" gap={1}>
                <HomeIcon fontSize="large" color="primary" />
                <Typography variant="body2">8418 NoodleHut, Good, WA, United States</Typography>
              </Stack>
              </Box>
          </Grid>
          <Grid item sm={12} md={4} display="flex" justifyContent={"center"}>
            <Box>
              <Typography variant="h5" align="center" color="primary">
                Hours
              </Typography>
              <Typography variant="body1">Tue - Fri:</Typography>
              <Typography variant="body2">11:00 AM - 3:00 PM & 4:30 PM - 8:30 PM</Typography>
              <Typography variant="body1">Sat:</Typography>
              <Typography variant="body2">12:00 PM - 3:00 PM & 4:30 PM - 8:30 PM</Typography>
              <Typography variant="body1">Mon & Sun</Typography>
              <Typography variant="body2">Closed</Typography>
            </Box>
          </Grid>

          <Grid item sm={12} md={4} display="flex" justifyContent={"center"}>
            <Box>
                <Typography variant="h5" component="div" align="center" color="primary">
                  Contact
                </Typography>
                <Stack
                  direction="row"
                  alignItems="center"
                  gap={1}
                  align="center"
                >
                  <ComputerIcon fontSize="large" color="primary" />
                  <Typography variant="body2">Developped by</Typography>
                  <Link to="https://github.com/mariered0">Marie H.</Link>
                </Stack>
            </Box>
          </Grid>
        </Grid>
      </Box>
    </Container>
  );
}

export default Home;
