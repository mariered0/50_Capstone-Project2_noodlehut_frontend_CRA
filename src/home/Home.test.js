import React from "react";
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import userContext from "../user/UserContext";
import Home from "./Home";

const currentUser = { username: 'test', 
                      firstName: 'Test', 
                      lastName: 'Test', 
                      phone: '0000000000', 
                      email: 'test@gmail.com', 
                      favorites: [], 
                      isAdmin: false };


it("renders without crashing", () => {
    render( 
            <MemoryRouter>
                <userContext.Provider value={{ currentUser }}>
                    <Home />
                </userContext.Provider>
            </MemoryRouter>
    );
});

it("matches snapshot", () => {
	const { asFragment } = render(
            <MemoryRouter>
                <userContext.Provider value={{ currentUser }}>
                    <Home />
                </userContext.Provider> 
            </MemoryRouter>
    );
	expect(asFragment()).toMatchSnapshot();
});