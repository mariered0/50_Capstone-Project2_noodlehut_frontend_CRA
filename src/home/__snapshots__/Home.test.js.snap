// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-n9njem-MuiContainer-root"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter MuiTypography-gutterBottom css-14v45o4-MuiTypography-root"
    >
      Welcome back, Test!
    </h3>
    <div
      class="MuiBox-root css-1l4w6pd"
    >
      <img
        alt="food image"
        class="MuiBox-root css-1yzt9ut"
        src="hero.jpg"
      />
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom title css-1gjpbx8-MuiTypography-root"
    >
      NoodleHut
    </h1>
    <p
      class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiTypography-paragraph css-17rfbpd-MuiTypography-root"
    >
      Thai home-style coocking. We serve authentically traditional Thai dishes.
      <br />
      No dine-in. This is a take-out restaurant.
    </p>
    <footer
      class="MuiBox-root css-byqtsl"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-spacing-md-5 css-2rm95u-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-507lne-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-y07eks-MuiTypography-root"
            >
              Visit
            </div>
            <div
              class="MuiStack-root css-1rlitzi-MuiStack-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-bsqp9w-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                8418 NoodleHut, Good, WA, United States
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-507lne-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-y07eks-MuiTypography-root"
            >
              Hours
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Tue - Fri:
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              11:00 AM - 3:00 PM & 4:30 PM - 8:30 PM
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Sat:
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              12:00 PM - 3:00 PM & 4:30 PM - 8:30 PM
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Mon & Sun
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              Closed
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-507lne-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-y07eks-MuiTypography-root"
            >
              Contact
            </div>
            <div
              align="center"
              class="MuiStack-root css-1rlitzi-MuiStack-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-bsqp9w-MuiSvgIcon-root"
                data-testid="ComputerIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
                />
              </svg>
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                Developped by
              </p>
              <a
                href="https://github.com/mariered0"
              >
                Marie H.
              </a>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
